tensorboard/webapp/header/header_component.ts

```diff
@@ -62,4 +71,12 @@ import {Component} from '@angular/core';
     `,
   ],
 })
-export class HeaderComponent {}
+export class HeaderComponent implements OnInit {
+  showUpload: boolean = false;
+
+  constructor(@Inject('window') private window: Window) {}
+
+  ngOnInit() {
+    this.showUpload = LOCAL_HOSTNAMES.includes(this.window.location.hostname);
```

Comment:
My preference would actually have been to push this logic into the tbdev-upload-button component while keeping the "margin" css value in this file.  However, I don't understand how I would be able to conditionally apply the "margin" in this file based on whether tbdev-upload-button implementation decides to render or not.

Reply:
Yes, you should definitely move this business logic into the upload-button file. We can talk about the margin offline.

Reply:
Offline you showed me how we could do this with @HostBinding.  So now the logic that determines whether to show the button is in the tbdev-upload-button component itself while the styling of margins when the button is shown still exists in the app-header.